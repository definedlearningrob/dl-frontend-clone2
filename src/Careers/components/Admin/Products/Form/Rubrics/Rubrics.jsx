import PropTypes from 'prop-types';
import { useField } from 'formik';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';

import { PaginationBar } from '@dc/components/Admin/Shared/List/PaginationBar/PaginationBar';
import RubricsDetailsModal from '@dc/components/Admin/Products/Form/Rubrics/Modal/Modal';
import { SortableAvailableList } from '@dc/components/Admin/Shared/SortableAvailableList/SortableAvailableList';
import { SortableSelectedList } from '@dc/components/Admin/Shared/SortableSelectedList/SortableSelectedList';
import { assignSteps } from '@dc/utils/assignSteps';

import SharedPaginatedLoader from '@shared/components/PaginatedLoader/PaginatedLoader';
import SharedLoadingSpinner from '@shared/components/LoadingSpinner/LoadingSpinner';
import Card from '@shared/components/Card/Card';
import { ListWrapper } from '@shared/components/SelectableList/ListWrapper/ListWrapper';
import { ReactComponent as TableIcon } from '@shared/svg/table.svg';
import Switch from '@shared/components/Switch/Switch';

AdminProductsFormRubrics.propTypes = {
  isWithCopies: PropTypes.bool,
  pagingProps: PropTypes.object,
  SearchBar: PropTypes.func,
  toggleIsWithCopies: PropTypes.func,
};

function AdminProductsFormRubrics({ pagingProps, SearchBar, toggleIsWithCopies, isWithCopies }) {
  const [activeProduct, setActiveProduct] = useState(null);
  const { t } = useTranslation();
  const [rubricsInput, , rubricsHelper] = useField('rubrics');
  const isDetailsModalOpen = activeProduct !== null;
  const sortedByStep = rubricsInput.value.slice().sort((a, b) => a.step - b.step);

  const closeDetailsModal = () => setActiveProduct(null);
  const openItemDetails = (product) => setActiveProduct(product);

  const removeItem = (item) => {
    const newTasks = rubricsInput.value.filter((existingItem) => {
      if (item.__typename) {
        return item.__typename !== existingItem.__typename || existingItem.id !== item.id;
      }

      return existingItem.id !== item.id;
    });

    rubricsHelper.setValue(assignSteps(newTasks));
  };
  const getKicker = (item) => {
    if (item.owner) {
      return {
        text: item.owner.name,
        type: 'secondary',
      };
    }
  };

  const handleEditClick = (rubric) => {
    window.open(`/admin/rubrics/${rubric.id}/edit`, '_blank', 'noreferrer');
  };

  return (
    <Card className='product-rubrics'>
      <h4>{t('admin.products.rubrics.label')}</h4>

      <div className='flex gap-sm items-center mr-xxs'>
        <ListWrapper
          actions={
            <div className='flex gap-xs items-center'>
              <Switch
                label={t('common.withCopies')}
                labelFirst={true}
                value={isWithCopies}
                onChange={toggleIsWithCopies}
              />
              <SearchBar
                field='name'
                placeholder={t('common.placeholders.searchBy', {
                  field: t('common.fields.common.name').toLowerCase(),
                })}
              />
            </div>
          }
          title={t('admin.products.rubrics.allRubrics')}>
          <div className='min-h-0 flex-1'>
            <SharedPaginatedLoader.Content
              SpinnerComponent={<SharedLoadingSpinner className='sortable-list-spinner' />}
              {...pagingProps}>
              {({ rubrics }) => (
                <SortableAvailableList
                  ListItemIcon={TableIcon}
                  field='rubrics'
                  getKicker={getKicker}
                  items={rubrics.nodes}
                  onDetailsOpen={openItemDetails}
                  onEditClick={handleEditClick}
                />
              )}
            </SharedPaginatedLoader.Content>
          </div>
          <PaginationBar pagingProps={pagingProps} />
        </ListWrapper>
        <ListWrapper title={`${t('common.statuses.selected')} (${sortedByStep.length})`}>
          <SortableSelectedList
            ListItemIcon={TableIcon}
            field='rubrics'
            getKicker={getKicker}
            isDragDisabled={true}
            items={sortedByStep}
            onChange={removeItem}
            onDetailsOpen={openItemDetails}
            onEditClick={handleEditClick}
          />
        </ListWrapper>
      </div>
      {isDetailsModalOpen && (
        <RubricsDetailsModal
          isOpen={isDetailsModalOpen}
          rubricId={activeProduct?.id}
          onClose={closeDetailsModal}
        />
      )}
    </Card>
  );
}

export default AdminProductsFormRubrics;
