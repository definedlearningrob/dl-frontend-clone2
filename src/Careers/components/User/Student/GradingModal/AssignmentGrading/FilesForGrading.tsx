import { useTranslation } from 'react-i18next';

import { ReactComponent as DownloadIcon } from '@dc/svg/download_to.svg';
import { ReactComponent as FileIcon } from '@dc/svg/file.svg';

import SharedIcon from '@shared/components/Icon/Icon';
import { ReactComponent as ShowIcon } from '@shared/svg/visible_outlined.svg';
import { CollapsibleWrapper } from '@shared/components/CollapsibleWrapper/CollapsibleWrapper';

type Props = {
  submission: {
    files:
      | {
          filename: string;
          url: string;
          previewUrl: string;
          id: string;
          attachmentUrl: string;
        }[];
  } | null;
};

export const FilesForGrading = ({ submission }: Props) => {
  const { t } = useTranslation();

  if (!submission) {
    return (
      <div
        className='text-secondary-500 p-sm text-center my-auto border'
        data-testid='unsubmitted-input'>
        {t('user.student.coursesActivity.assignmentNoInputInfo')}
      </div>
    );
  }

  return (
    <CollapsibleWrapper title={t('user.student.coursesActivity.files')}>
      <ul className='mb-base'>
        {submission.files.map((file) => (
          <li
            key={file.id}
            className='user-student__courses-activity-modal__assignment-file-item'
            data-testid='submission-file-item'>
            <a href={file.url} rel='noopener noreferrer' target='_blank'>
              <SharedIcon icon={<FileIcon />} size='sm' />
              <span className='user-student__courses-activity-modal__assignment-file-name'>
                {file.filename}
              </span>
            </a>
            <div className='user-student__courses-activity-modal__assignment-file-buttons'>
              <a
                href={file.previewUrl}
                rel='noopener noreferrer'
                target='_blank'
                onClick={(e) => e.stopPropagation()}>
                <SharedIcon icon={<ShowIcon />} size='sm' />
              </a>
              <a
                download={file.filename}
                href={file.attachmentUrl}
                rel='noopener noreferrer'
                target='_blank'
                onClick={(e) => e.stopPropagation()}>
                <SharedIcon icon={<DownloadIcon />} size='sm' />
              </a>
            </div>
          </li>
        ))}
      </ul>
    </CollapsibleWrapper>
  );
};
