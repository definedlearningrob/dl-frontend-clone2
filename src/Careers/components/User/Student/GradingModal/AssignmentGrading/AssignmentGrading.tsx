import { TRubric } from '@dc/graphql/user/queries/lesson';
import { FilesForGrading } from '@dc/components/User/Student/GradingModal/AssignmentGrading/FilesForGrading';

import { cleanInjection } from '@shared/utils/cleanInjection';
import { RubricsViewer } from '@shared/components/Rubrics/RubricsViewer';
import { CollapsibleWrapper } from '@shared/components/CollapsibleWrapper/CollapsibleWrapper';

type TStudentAssignment = {
  description: string;
  displayName: string;
  rubrics: TRubric[];
  id: string;
  submission: {
    files:
      | {
          filename: string;
          url: string;
          previewUrl: string;
          id: string;
          attachmentUrl: string;
        }[];
  } | null;
};

type Props = {
  assignment: TStudentAssignment;
};

export const AssignmentGrading = ({ assignment }: Props) => {
  const { description, displayName, submission } = assignment;

  return (
    <div className='h-full flex flex-col'>
      <CollapsibleWrapper title={displayName}>
        <div
          className='injected-content pt-sm'
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={cleanInjection(description)}
        />
      </CollapsibleWrapper>
      <FilesForGrading submission={submission} />
      {submission && <RubricsViewer />}
    </div>
  );
};
