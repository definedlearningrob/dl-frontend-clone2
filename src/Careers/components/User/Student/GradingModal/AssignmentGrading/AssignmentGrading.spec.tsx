import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { renderWithI18N } from '@dc/utils/test';
import { AssignmentGrading } from '@dc/components/User/Student/GradingModal/AssignmentGrading/AssignmentGrading';

import { RubricProvider } from '@shared/components/Rubrics/RubricProvider';
import { RUBRIC_TYPE } from '@shared/components/Rubrics/utils/enums';
import { defaultRubricData } from '@shared/components/Rubrics/utils/defaultRubricData';

const assignment = {
  displayName: 'Assignment name',
  description: 'Assignment description',
  rubrics: [],
  id: '1',
  submission: null,
};

const defaultRubric = { ...defaultRubricData, id: '1', name: 'Default name', pointsAvailable: 10 };

describe('AssignmentGrading', () => {
  it('should render correctly without submission', () => {
    const { container } = renderWithI18N(
      <RubricProvider rubric={defaultRubric} type={RUBRIC_TYPE.GRADER}>
        <AssignmentGrading assignment={assignment} />
      </RubricProvider>
    );

    expect(screen.getByText('Assignment name')).toBeInTheDocument();
    expect(screen.getByText('Student has not submitted any files yet.')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with submission', () => {
    const assignmentWithSubmission = {
      ...assignment,
      submission: {
        files: [
          {
            id: '1',
            filename: 'filename',
            url: 'url',
            previewUrl: 'previewUrl',
            attachmentUrl: 'attachmentUrl',
          },
          {
            id: '2',
            filename: 'filename 2',
            url: 'url2',
            previewUrl: 'previewUrl2',
            attachmentUrl: 'attachmentUrl2',
          },
        ],
      },
    };
    const { container } = renderWithI18N(
      <RubricProvider rubric={defaultRubric} type={RUBRIC_TYPE.GRADER}>
        <AssignmentGrading assignment={assignmentWithSubmission} />
      </RubricProvider>
    );

    userEvent.click(screen.getByText('Files'));

    expect(screen.getByText('filename')).toBeInTheDocument();
    expect(screen.getByText('filename 2')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
