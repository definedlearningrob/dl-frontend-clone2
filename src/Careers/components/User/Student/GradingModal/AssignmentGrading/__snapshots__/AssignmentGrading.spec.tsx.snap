// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignmentGrading should render correctly with submission 1`] = `
<div>
  <div
    class="h-full flex flex-col"
  >
    <div
      class="group/statement relative px-sm py-xs xxxl:p-base xxxl:py-sm bg-white rounded-sm transition-all cursor-pointer border border-neutral-300 mb-xs hover:bg-neutral-200 hover:border-neutral-400"
    >
      <h3
        class="text-sm flex items-center justify-between mb-0"
      >
        Assignment name
        <button
          class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 w-md h-md"
          data-testid="icon-button"
          type="button"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-base w-base"
              />
            </div>
          </div>
        </button>
      </h3>
    </div>
    <div
      class="group/statement relative px-sm py-xs xxxl:p-base xxxl:py-sm bg-white rounded-sm transition-all cursor-pointer border mb-xs border-primary-500 mb-sm xxxl:mb-x pb-base xxxl:pb-md"
    >
      <h3
        class="text-sm flex items-center justify-between mb-0"
      >
        Files
        <button
          class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 w-md h-md"
          data-testid="icon-button"
          type="button"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-base w-base"
              />
            </div>
          </div>
        </button>
      </h3>
      <ul
        class="mb-base"
      >
        <li
          class="user-student__courses-activity-modal__assignment-file-item"
          data-testid="submission-file-item"
        >
          <a
            href="url"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="icon -sm-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="user-student__courses-activity-modal__assignment-file-name"
            >
              filename
            </span>
          </a>
          <div
            class="user-student__courses-activity-modal__assignment-file-buttons"
          >
            <a
              href="previewUrl"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="icon -sm-size"
                data-testid="icon"
              >
                <div />
              </div>
            </a>
            <a
              download="filename"
              href="attachmentUrl"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="icon -sm-size"
                data-testid="icon"
              >
                <div />
              </div>
            </a>
          </div>
        </li>
        <li
          class="user-student__courses-activity-modal__assignment-file-item"
          data-testid="submission-file-item"
        >
          <a
            href="url2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="icon -sm-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="user-student__courses-activity-modal__assignment-file-name"
            >
              filename 2
            </span>
          </a>
          <div
            class="user-student__courses-activity-modal__assignment-file-buttons"
          >
            <a
              href="previewUrl2"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="icon -sm-size"
                data-testid="icon"
              >
                <div />
              </div>
            </a>
            <a
              download="filename 2"
              href="attachmentUrl2"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="icon -sm-size"
                data-testid="icon"
              >
                <div />
              </div>
            </a>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="overflow-x-auto w-full"
    >
      <table
        class="bg-white border-separate border-spacing-0 rounded-xs w-full table-fixed"
        data-testid="rubric-table-builder"
      >
        <thead>
          <tr>
            <th
              class="border border-neutral-300 p-xs xxxl:p-x text-left text-xs font-bold bg-neutral-200 w-[160px]"
              data-testid="rubrics-reserved-cell"
            >
              Levels
            </th>
            <th
              class="border border-neutral-300 p-xs xxxl:p-x text-xs font-bold bg-neutral-200"
              data-testid="rubric-criteria-label"
            >
              <div
                class="flex flex-col"
              >
                <span
                  data-testid="rubric-criteria-label-score"
                >
                  1
                </span>
                <span
                  data-state="closed"
                >
                  <div
                    class="line-clamp-2 xxxl:line-clamp-3"
                    data-testid="rubric-criteria-label-text"
                  >
                    test label
                  </div>
                </span>
              </div>
            </th>
            <th
              class="border border-neutral-300 p-xs xxxl:p-x text-xs font-bold bg-neutral-200"
              data-testid="rubric-criteria-label"
            >
              <div
                class="flex flex-col"
              >
                <span
                  data-testid="rubric-criteria-label-score"
                >
                  2
                </span>
              </div>
            </th>
            <th
              class="border border-neutral-300 p-xs xxxl:p-x text-xs font-bold bg-neutral-200"
              data-testid="rubric-criteria-label"
            >
              <div
                class="flex flex-col"
              >
                <span
                  data-testid="rubric-criteria-label-score"
                >
                  3
                </span>
              </div>
            </th>
            <th
              class="border border-neutral-300 p-xs xxxl:p-x text-xs font-bold bg-neutral-200"
              data-testid="rubric-criteria-label"
            >
              <div
                class="flex flex-col"
              >
                <span
                  data-testid="rubric-criteria-label-score"
                >
                  4
                </span>
              </div>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="border border-neutral-300 p-xs xxxl:p-x"
              data-testid="rubric-heading"
            >
              <span
                data-state="closed"
              >
                <div
                  class="text-xs xxxl:text-sm font-bold text-neutral-800 mb-xs line-clamp-4"
                  data-testid="rubric-heading-text"
                >
                  rubric.careerPathway.name
                </div>
              </span>
              <span
                class="leading-lg text-xxs xxxl:text-xs me-xxs"
              >
                Multiplier:
              </span>
              <span
                class="text-neutral-800 leading-lg text-xxs xxxl:text-xs px-xxs border border-neutral-300 min-w-fit rounded-xs"
                data-testid="rubric-heading-multiplier"
              >
                x1
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.careerPathway.1
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.careerPathway.2
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.careerPathway.3
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.careerPathway.4
              </span>
            </td>
          </tr>
          <tr>
            <td
              class="border border-neutral-300 p-xs xxxl:p-x"
              data-testid="rubric-heading"
            >
              <span
                data-state="closed"
              >
                <div
                  class="text-xs xxxl:text-sm font-bold text-neutral-800 mb-xs line-clamp-4"
                  data-testid="rubric-heading-text"
                >
                  rubric.projectGoal.name
                </div>
              </span>
              <span
                class="leading-lg text-xxs xxxl:text-xs me-xxs"
              >
                Multiplier:
              </span>
              <span
                class="text-neutral-800 leading-lg text-xxs xxxl:text-xs px-xxs border border-neutral-300 min-w-fit rounded-xs"
                data-testid="rubric-heading-multiplier"
              >
                x1
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.projectGoal.1
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.projectGoal.2
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.projectGoal.3
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.projectGoal.4
              </span>
            </td>
          </tr>
          <tr>
            <td
              class="border border-neutral-300 p-xs xxxl:p-x"
              data-testid="rubric-heading"
            >
              <span
                data-state="closed"
              >
                <div
                  class="text-xs xxxl:text-sm font-bold text-neutral-800 mb-xs line-clamp-4"
                  data-testid="rubric-heading-text"
                >
                  rubric.projectRole.name
                </div>
              </span>
              <span
                class="leading-lg text-xxs xxxl:text-xs me-xxs"
              >
                Multiplier:
              </span>
              <span
                class="text-neutral-800 leading-lg text-xxs xxxl:text-xs px-xxs border border-neutral-300 min-w-fit rounded-xs"
                data-testid="rubric-heading-multiplier"
              >
                x1
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.projectRole.1
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.projectRole.2
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.projectRole.3
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.projectRole.4
              </span>
            </td>
          </tr>
          <tr>
            <td
              class="border border-neutral-300 p-xs xxxl:p-x"
              data-testid="rubric-heading"
            >
              <span
                data-state="closed"
              >
                <div
                  class="text-xs xxxl:text-sm font-bold text-neutral-800 mb-xs line-clamp-4"
                  data-testid="rubric-heading-text"
                >
                  rubric.jobProductQuality.name
                </div>
              </span>
              <span
                class="leading-lg text-xxs xxxl:text-xs me-xxs"
              >
                Multiplier:
              </span>
              <span
                class="text-neutral-800 leading-lg text-xxs xxxl:text-xs px-xxs border border-neutral-300 min-w-fit rounded-xs"
                data-testid="rubric-heading-multiplier"
              >
                x1
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.jobProductQuality.1
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.jobProductQuality.2
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.jobProductQuality.3
              </span>
            </td>
            <td
              class="group/criteria border border-neutral-300 p-xs xxxl:p-x align-top leading-lg focus:border focus:border-primary-500 hover:bg-neutral-200"
              data-testid="rubric-criteria"
              role="button"
              tabindex="1"
            >
              <div
                class="flex justify-between items-center mb-xs xxxl:mb-x relative"
              >
                <label
                  class="radioLabel"
                >
                  <input
                    class="radioInput"
                    readonly=""
                    tabindex="1"
                    type="radio"
                    value=""
                  />
                  <div
                    class="radioSpan"
                  >
                    <span
                      class="labelWrapper"
                    />
                  </div>
                </label>
              </div>
              <span
                class="text-xxs xxxl:text-xs tracking-[0.12px] text-neutral-700"
                data-testid="rubric-criteria-text"
              >
                rubric.jobProductQuality.4
              </span>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="pt-md basis-full"
      />
    </div>
  </div>
</div>
`;

exports[`AssignmentGrading should render correctly without submission 1`] = `
<div>
  <div
    class="h-full flex flex-col"
  >
    <div
      class="group/statement relative px-sm py-xs xxxl:p-base xxxl:py-sm bg-white rounded-sm transition-all cursor-pointer border border-neutral-300 mb-xs hover:bg-neutral-200 hover:border-neutral-400"
    >
      <h3
        class="text-sm flex items-center justify-between mb-0"
      >
        Assignment name
        <button
          class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 w-md h-md"
          data-testid="icon-button"
          type="button"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-base w-base"
              />
            </div>
          </div>
        </button>
      </h3>
    </div>
    <div
      class="text-secondary-500 p-sm text-center my-auto border"
      data-testid="unsubmitted-input"
    >
      Student has not submitted any files yet.
    </div>
  </div>
</div>
`;
